Modules/signalmodule.c

```diff
@@ -1346,21 +1346,19 @@ PyInit__signal(void)
     initialized = 1;
 #endif
 
-    /* Add some symbolic constants to the module */
     d = PyModule_GetDict(m);
 
     x = DefaultHandler = PyLong_FromVoidPtr((void *)SIG_DFL);
-    if (!x || PyDict_SetItemString(d, "SIG_DFL", x) < 0)
-        goto finally;
+    if (PyModule_AddObject(m, "SIG_DFL", x))
+         goto finally;
 
     x = IgnoreHandler = PyLong_FromVoidPtr((void *)SIG_IGN);
-    if (!x || PyDict_SetItemString(d, "SIG_IGN", x) < 0)
-        goto finally;
+    if (PyModule_AddObject(m, "SIG_IGN", x))
+         goto finally;
 
     x = PyLong_FromLong((long)NSIG);
-    if (!x || PyDict_SetItemString(d, "NSIG", x) < 0)
-        goto finally;
-    Py_DECREF(x);
+    if (PyModule_AddObject(m, "NSIG", x))
```

Comment:
I think we can use `PyModule_AddIntMacro(m, NSIG)` directly here.

Reply:
This was complaining with type errors that is why I changed. Does this PyModule_AddIntMacro(m, NSIG) give you a green build on your side?

Reply:
Yes, I can compile signalmodule.c without any warnings on macOS. Here's my patch:

```diff
diff --git a/Modules/signalmodule.c b/Modules/signalmodule.c
index 8690897055..b1c4d95637 100644
--- a/Modules/signalmodule.c
+++ b/Modules/signalmodule.c
@@ -1356,8 +1356,7 @@ PyInit__signal(void)
     if (PyModule_AddObject(m, "SIG_IGN", x))
          goto finally;

-    x = PyLong_FromLong((long)NSIG);
-    if (PyModule_AddObject(m, "NSIG", x))
+    if (PyModule_AddIntMacro(m, NSIG))
          goto finally;

 #ifdef SIG_BLOCK
```

Reply:
Let me confirm on my side and update accordingly.